app/models/person.rb

```diff
@@ -37,6 +37,7 @@ def downcase_diaspora_handle
 
   has_many :contacts, :dependent => :destroy # Other people's contacts for this person
   has_many :posts, :foreign_key => :author_id, :dependent => :destroy # This person's own posts
+  has_one  :pinned_post, -> { where(pinned: true) }, :foreign_key => :author_id, :dependent => :destroy, :class_name => "Post" # This person's pinned post
```

Comment:
But then it will be `belongs_to :pinnded_post`, right? What do you think about adding a model validation instead?

Reply:
A validation would be a solution, but I still think that something that just can't go wrong is better, so we don't rely on validations (validations don't work when things are received in parallel). Also it makes updating easier, because you can just set a new ID and don't need to remove the `pinned` flag from the old post.